### DISCUSSION TITLE
Add an option to bundle all external dependencies

### DISCUSSION THREAD

**Discussion by srfsh** [CaiJimmy/hugo-theme-stack#1031] [Ideas]
Some people might not prefer to have external dependencies on their websites. I think this is truly a great future to have. So I thought I should raise this. What are your opinions about it?

**Comment by louxyy** [comment]
Which dependency are you thinking of ?

**Reply by srfsh** [reply]
Excuse the late reply.

The ones that depend on external services, as mentioned.  There exists a dependency on jsdelivr.net, for instance.

**Reply by ElastiCourse** [reply]
> Excuse the late reply.
> 
> The ones that depend on external services, as mentioned. There exists a dependency on jsdelivr.net, for instance.

@FarisZR built this amazing script for that https://github.com/FarisZR/Hugo-stack-no-cdn

Reference original issue: https://github.com/CaiJimmy/hugo-theme-stack/issues/263

**Reply by FarisZR** [reply]
Thanks for the mention, however I'm not sure if it's still up-to-date or not.
I'm still using it on my site though

**Reply by srfsh** [reply]
> @FarisZR built this amazing script for that https://github.com/FarisZR/Hugo-stack-no-cdn
> 
> Reference original issue: #263

I couldn't figure out how would one use this. Is there a manual of some sort?

**Reply by ElastiCourse** [reply]
> > @FarisZR built this amazing script for that https://github.com/FarisZR/Hugo-stack-no-cdn
> > Reference original issue: #263
> 
> I couldn't figure out how would one use this. Is there a manual of some sort?

This setup is using Github actions for automating the process. If you are using Github to host your blog code, you can use this feature for free to run automation pipelines in your repo.

If you are looking for an easy way to implement this, without having to go through this automation process, here are the steps:
1. Download the `hugo-stack-no-cdn` files as ZIP here https://github.com/FarisZR/Hugo-stack-no-cdn/archive/refs/heads/main.zip
2. Extract the ZIP file
3. Copy both `data` and `static` folders into your blog home folder, you should be asked if you want to override `external.yaml`, choose Yes.
4. Generate your new blog using `hugo --gc --minify` command, or trigger a deploy using whatever service you are setup with (Netlify, Github pages, Vercel.. etc)

**Reply by FarisZR** [reply]
There is a much easier way, Hugo modules,  just add it to your config file and then `hugo mod get -u`
